src/libraries/System.Formats.Tar/tests/TarWriter/TarWriter.WriteEntry.File.Tests.Unix.cs

```diff
@@ -139,5 +139,40 @@ public void Add_CharacterDevice(TarEntryFormat format)
 
             }, format.ToString(), new RemoteInvokeOptions { RunAsSudo = true }).Dispose();
         }
+
+        [ConditionalFact(nameof(IsRemoteExecutorSupportedAndPrivilegedProcess))]
+        public void CreateEntryFromFileOwnedByNonExistentGroup()
+        {
+            RemoteExecutor.Invoke(() =>
+            {
+                string groupName = Path.GetRandomFileName()[0..6];
+                int groupId = CreateGroup(groupName);
+
+                using TempDirectory root = new TempDirectory();
+
+                string fileName = "file.txt";
+                string filePath = Path.Join(root.Path, fileName);
+                File.Create(filePath).Dispose();
+
+                SetGroupAsOwnerOfFile(groupName, filePath);
+
+                DeleteGroup(groupName);
```

Comment:
we can delay the creation of the group to a moment before we need to use it for the first time. So when creation of the file fails, we don't create the group and don't need to delete it. Then move the removal to finally block so we always try to delete it and increase the chances of not having too many ghost groups on the CI machines.

```suggestion
                using TempDirectory root = new TempDirectory();

                string fileName = "file.txt";
                string filePath = Path.Join(root.Path, fileName);
                File.Create(filePath).Dispose();

                string groupName = Path.GetRandomFileName()[0..6];
                int groupId = CreateGroup(groupName);

                try
                {
                    SetGroupAsOwnerOfFile(groupName, filePath);
                }
                finally
                {
                    DeleteGroup(groupName);
                }
```